boards/PDM/Src/App/states/App_DriveState.c

```diff
@@ -0,0 +1,192 @@
+    }
+    else if (App_Efuse_FaultProcedure_Channel1(efuse4, 3) == 0)
+    {
+        App_CanTx_PDM_Efuse_Fault_Checks_FAN_Set(true);
+    }
+}
+
+bool Fault_Detection(
+    struct Efuse *         efuse1,
+    struct Efuse *         efuse2,
+    struct Efuse *         efuse3,
+    struct Efuse *         efuse4,
+    struct RailMonitoring *rail_monitor)
+{
+    bool status = false;
+    if (App_Efuse_FaultProcedure_Channel0(efuse1, 3) == 2)
+    {
+        // AIR
+        App_CanTx_PDM_Efuse_Fault_Checks_AIR_Set(true);
+        status = true;
+    }
+    if (App_Efuse_FaultProcedure_Channel0(efuse2, 0) == 2)
+    {
+        // EMETER
+        App_CanTx_PDM_Efuse_Fault_Checks_EMETER_Set(true);
+        status = true;
+    }
+    if (App_Efuse_FaultProcedure_Channel0(efuse3, 0) == 2)
+    {
+        // LEFT INVERTER
+        App_CanTx_PDM_Efuse_Fault_Checks_LEFT_INVERTER_Set(true);
+        status = true;
+    }
+    if (App_Efuse_FaultProcedure_Channel1(efuse3, 0) == 2)
+    {
+        // LEFT INVERTER
+        App_CanTx_PDM_Efuse_Fault_Checks_RIGHT_INVERTER_Set(true);
+        status = true;
+    }
+    if (App_RailMonitoring_VBAT_VoltageCriticalCheck(rail_monitor))
+    {
+        // VBAT
+        // NO CAN MESSAGE BECAUSE IT'S IN ALLSTATES
+        status = true;
+    }
+    if (App_RailMonitoring__24V_ACC_VoltageCriticalCheck(rail_monitor))
+    {
+        // ACC
+        // NO CAN MESSAGE BECAUSE IT'S IN ALLSTATES
+        status = true;
+    }
+    if (App_RailMonitoring__22V_AUX_VoltageCriticalCheck(rail_monitor))
+    {
+        // AUX
+        // NO CAN MESSAGE BECAUSE IT'S IN ALLSTATES
+        status = true;
+    }
+
+    return status;
+}
+
+static void DriveStateRunOnTick1Hz(struct StateMachine *const state_machine)
+{
+    App_AllStatesRunOnTick1Hz(state_machine);
+}
+
+static void DriveStateRunOnTick100Hz(struct StateMachine *const state_machine)
+{
+    struct PdmWorld *      world        = App_SharedStateMachine_GetWorld(state_machine);
+    struct RailMonitoring *rail_monitor = App_PdmWorld_GetRailMonitoring(world);
+    struct Efuse *         efuse1       = App_PdmWorld_GetEfuse1(world);
+    struct Efuse *         efuse2       = App_PdmWorld_GetEfuse2(world);
+    struct Efuse *         efuse3       = App_PdmWorld_GetEfuse3(world);
+    struct Efuse *         efuse4       = App_PdmWorld_GetEfuse4(world);
+    bool                   has_fault    = Fault_Detection(efuse1, efuse2, efuse3, efuse4, rail_monitor);
+
+    App_AllStatesRunOnTick100Hz(state_machine);
+    if (!has_fault)
+    {
+        Warning_Detection(efuse1, efuse2, efuse3, efuse4, rail_monitor);
+
+        if (1 == 1) // To be replaced with App_CanTx_BMSStates_AIRNEGATIVE closed
+        {
+            // PCM Running
+            Efuse_Enable_Channels_PCM_RUNNING(efuse1, efuse2, efuse3, efuse4);
+        }
+        else
+        {
+            // 18650 Startup
+            Efuse_Enable_Channels_18650_STARTUP(efuse1, efuse2, efuse3, efuse4);
+        }
+    }
```

Comment:
Not sure if this logic works. If we're in Drive state, isn't the PCM already running? Does startup occur in Init state? Looks like state machine behavior is unfinished rn

Reply:
Also won't all faults trip before startup, because current is low due to efuses being disabled?